client/src/main/java/io/pravega/client/netty/impl/FlowHandler.java

```diff
@@ -43,16 +44,16 @@
     private final AtomicBoolean recentMessage = new AtomicBoolean(false);
     private final AtomicBoolean closed = new AtomicBoolean(false);
     @Getter
-    private final AppendBatchSizeTracker batchSizeTracker;
     private final ReusableFutureLatch<Void> registeredFutureLatch = new ReusableFutureLatch<>();
     @VisibleForTesting
     @Getter(AccessLevel.PACKAGE)
     private final ConcurrentHashMap<Integer, ReplyProcessor> flowIdReplyProcessorMap = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<UUID, AppendBatchSizeTracker> idBatchSizeTrackerMap = new ConcurrentHashMap<>();
```

Comment:
We can use `ConcurrentHashMap<Integer, AppendBatchSizeTracker> idBatchSizeTrackerMap` to maitain a map between the segment writer and its corresponding AppendBatchSizeTracker.

Reply:
@shrids Can we make that integer into an object? Possibly by defining a equals and hashcode on flow or having a flowid? I'd rather avoid a primitive as it is non-obvious what it means/is.

Reply:
yes, we can make it into an object which represents the FlowId here.